nodes/src/nodes/approval/approval_manager.py

```
@@ -0,0 +1,218 @@
+        Returns:
+            The updated approval request dict.
+
+        Raises:
+            KeyError: If *approval_id* is not found.
+            ValueError: If the request is not in 'pending' status.
+        """
+        with self._lock:
+            request = self._get_request(approval_id)
+            self._check_timeout(approval_id)
+            if request['status'] != 'pending':
+                raise ValueError(f'Cannot approve request in status: {request["status"]}')
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)